python/python/Ice/Properties.py

```diff
@@ -13,8 +13,59 @@ class Properties:
     application-name[.category[.sub-category]].name.
     """
 
-    def __init__(self, impl: IcePy.Properties):
-        self._impl = impl
+    @overload
+    def __init__(self, args: list[str] | None = None, defaults: Self | None = None): ...
+
+    @overload
+    def __init__(self, *, properties: IcePy.Properties): ...
```

Comment:
Do we need this overload? I assume we only call this constructor from the C++ code.